pkg/pipelinerunmetrics/metrics.go

```diff
@@ -272,9 +294,21 @@ func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
 	}
 
 	var runningPRs int
+	var trsWaitResolvingTaskRef int
+	var prsWaitResolvingPipelineRef int
+
 	for _, pr := range prs {
 		if !pr.IsDone() {
 			runningPRs++
+			succeedCondition := pr.Status.GetCondition(apis.ConditionSucceeded)
+			if succeedCondition != nil && succeedCondition.Status == corev1.ConditionUnknown {
+				switch succeedCondition.Reason {
+				case v1.TaskRunReasonResolvingTaskRef:
```

Comment:
Does this occur if the pipelinerun has status v1.TaskRunReasonResolvingTaskRef? I'm a bit confused how the PR status is updated if it's waiting on multiple taskrun resolutions. Wouldn't the TR metric be inaccurate in this case, because it's incremented once per pipelinerun rather than once per taskrun in each pipelinerun?

Reply:
So the answer to your first question is "yes".

Specifically, the code at https://github.com/tektoncd/pipeline/blob/c88734738864832d67bd9e8629bd5a1dc7d387bd/pkg/reconciler/pipelinerun/pipelinerun.go#L538-L547 is what motivated adding this metric.

Now, looking at the implementation of `resolvePipelineState` it errors out of that method on the first instance of `remote.ErrRequestInProgress` .... so it does not bother to see how many taskruns it is waiting on.

From at least my team's operational perspective, knowing that a pipelinesrun is blocked by the resolution of *any* of its taskruns is sufficient..... i.e. the number of pipelineruns blocked by taskruns does not have to equal the number of blocked taskruns.

Lastly, I looked at my description for the metric, and I'll admit it is not precisely clear on this nuance (i.e. I need the throw the word "any" in there in the right spot).

Hopefully that helps with the bit you said you were confused on and answers the second question, and of course if you agree with my logic/thinking here.  And if you think I need to put some form of all this ^^ in comments in the code, or in the metrics.md file let me know.

Thansk

Reply:
Ok I think this makes more sense now, thank you! I was thinking that it was supposed to reflect the number of blocked taskruns in a pipelinerun, but re-reading the metric names/descriptions this is more clear. I'm wondering if it's useful to separate out waiting on pipeline resolution vs waiting on task resolution, vs just the number of PRs waiting on resolution of any kind? Genuine question, curious to hear what is useful for you.

Reply:
Tag related to resolution can be helpful but this should be configurable via configmap. We don't want to increase the cardinality.

Reply:
Ack on the resolution/configmap note @khrm ..... I'll make sure and including the configuration / opt in approach for those labels if and when I come back with a new PR that contributes that.

On the waiting for resolution of any other kind, I think the reason label we check for is for any of the resolver types.  Though I've referenced image/bundle resolution in some of my comments here, the other types are not precluded.  And I was careful in the metric description to not reference a specific resolver.  Certainly though if we were to add configurable labels in a follow up PR, specific configurable labels for each of the types could make sense.  Not adding the more detailed labels with this initial PR bypasses all that additional code delta.